app/assets/javascripts/discourse/app/components/topic-navigation-popup.js

```diff
@@ -1,12 +1,32 @@
 import Component from "@ember/component";
 import { action } from "@ember/object";
+import discourseComputed from "discourse-common/utils/decorators";
 
 export default Component.extend({
   tagName: "",
   hidden: false,
 
+  didInsertElement() {
+    this._super(...arguments);
+
+    if (this.noticeKey && this.keyValueStore.getItem(this.noticeKey)) {
+      this.set("hidden", true);
+    }
+  },
```

Comment:
Putting this in `didInsertElement` will mean that we get the sequence:

1. Ember renders template with hidden=false
2. `didInsertElement` run
3. Ember now has to re-render the template

Can we put this logic in the `init()` hook? Or maybe `didReceiveAttrs`?